### PR TITLE
[BUGFIX] Ensure `test_script_runner.py` actually raises AssertionErrors correctly

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- Fixed bug in `test_script_runner.py` which caused `AssertionErrors` (along with other errors) to be silently logged instead of being raised. 
- Fixed bug in `test_script_runner.py` so that `tmp_directory` and `base_directory` are navigated explicitly. This prevents errors in the integration tests where the GE package was not being installed correctly. 
- Updated 2 tests that came from the bugfix. 
-  For `multi_batch_rule_based_profiler_example.py`, there is a JIRA task GREAT-594 to improve the ergonomics so at users dont have to refer to min and max values by their indices. 

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: tests/integration/test_script_runner.py [hunk 1]
```diff
@@ -712,9 +712,10 @@ def _execute_integration_test(integration_test_fixture, tmp_path):
     workdir = os.getcwd()
     try:
         base_dir = file_relative_path(__file__, "../../")
-        os.chdir(tmp_path)
+        os.chdir(base_dir)
         # Ensure GE is installed in our environment
         execute_shell_command("pip install .")
+        os.chdir(tmp_path)
```
Comment (Shinnnyshinshin) [inline tests/integration/test_script_runner.py:718]: Need to explicitly change to the `base_dir` (ie. where `great_expectations/` is) in order to do the installation before switching back to our current `tmp_path`

File: tests/integration/test_script_runner.py [hunk 2]
```diff
@@ -774,6 +775,7 @@ def _execute_integration_test(integration_test_fixture, tmp_path):
         loader.exec_module(test_script_module)
     except Exception as e:
         logger.error(str(e))
+        raise
```
Comment (Shinnnyshinshin) [inline tests/integration/test_script_runner.py:778]: This is the line that was causing the issue

Comment (NathanFarmer) [reply tests/integration/test_script_runner.py:778]: ❤️

File: tests/integration/docusaurus/expectations/advanced/multi_batch_rule_based_profiler_example.py [hunk 3]
```diff
@@ -37,8 +37,8 @@
       - expectation_type: expect_table_row_count_to_be_between
         class_name: DefaultExpectationConfigurationBuilder
         module_name: great_expectations.rule_based_profiler.expectation_configuration_builder
-        min_value: $parameter.row_count_range.value.min_value
```
Comment (NathanFarmer) [inline tests/integration/docusaurus/expectations/advanced/multi_batch_rule_based_profiler_example.py:40]: Just confirming - this was already broken, but was just getting skipped?

**Review by NathanFarmer — APPROVED** [review]
LGTM!

File: tests/integration/docusaurus/expectations/advanced/multi_batch_rule_based_profiler_example.py (hunk 3 above)
Comment (Shinnnyshinshin) [reply tests/integration/docusaurus/expectations/advanced/multi_batch_rule_based_profiler_example.py:40]: Yes it was broken, but wasn't getting detected.

**Pull request merged**